temporian/core/test/compilation_test.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List, Tuple
+from absl.testing import absltest
+
+import temporian as tp
+from temporian.implementation.numpy.data.event_set import EventSet
+
+
+class CompileTest(absltest.TestCase):
```

Comment:
Could you also test the following situation? (to make sure a full and single graph is traced).
Note: To test that a graph is traced exactly once, you can instrument the traced functions.

1.
@tf.compile
def f(x)
    return tp.something(x)

@tf.compile
def g(x)
    return f(x) +x

2.
@tf.compile
def f(x, b:bool)
    if b:
        return tp.something(x)
    else:
        return tp.somethingelse(x)
3.
@tf.compile
def f(x, b:int)
    for _ in range(b):
        x = tp.something(x)
    return x

4.
(If possible)

class A:
    param:int = 5

    @tf.compile
    def f(x)
        return tp.something(x)

    @tf.compile
    def g(self, x)
        return tp.something(x, self.param)

Reply:
added TODO